lib/audited/auditor.rb

```diff
@@ -247,12 +247,11 @@ def audited_changes(for_touch: false)
             all_changes.except(*self.class.non_audited_columns)
           end
 
-        if for_touch
+        if for_touch && audits.present?
+          last_audit = audits.last.audited_changes
```

Comment:
Another small optimization. No need to perform 2 queries - for the `#present?` and to get the last one. If there are no audits, then the query for the last one would result in a `nil`.

```suggestion
        if for_touch && (last_audit = audits.last)
          changes = last_audit.audited_changes
```